src/cached_object_store.rs

```diff
@@ -254,7 +257,7 @@ impl CachedObjectStore {
             let bytes = get_result.bytes().await?;
             entry.save_head((&meta, &attrs)).await.ok();
             entry.save_part(part_id, bytes.clone()).await.ok();
-            Ok(bytes.slice(range_in_part))
+            Ok(bytes.slice(range_in_part).clone())
```

Comment:
I believe `clone` just adds another reference, so this still has the same problem. We'd need to do something like:
`Bytes::copy_from_slice(bytes.slice(range_in_part).as_ref())`

Reply:
oh no, you're right:

```
impl Clone for Bytes {
    #[inline]
    fn clone(&self) -> Bytes {
        unsafe { (self.vtable.clone)(&self.data, self.ptr, self.len) }
    }
}

```

fixed it in 8b53ec4